airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java

```diff
@@ -152,8 +152,19 @@ private static boolean isCheckWithCatalog(IntegrationConfig parsed) {
 
   private AirbyteMessage runCheckWithCatalog(final IntegrationConfig parsed) throws Exception {
     LOGGER.info("Using Check With Catalog Method");
-    // TODO: Actually implement something different
-    return runCheck(parsed);
+    if (!(integration instanceof Check)) {
+      LOGGER.warn("Catalog Sent to check which does not implement checkWithCatalog for connector {}", integration);
+      return runCheck(parsed);
```

Comment:
it's also a bit confusing potentially to future readers that this is runCheck when it's not a Check.. Can this method be renamed?

Reply:
done